docs/reference/src/main/asciidoc/events.asciidoc

```diff
@@ -93,25 +100,70 @@ injection:
 @Inject @Any Event<Document> documentEvent;
 -------------------------------------------
 
-A producer raises events by calling the `fire()` method of the `Event`
+==== Synchronous event producers
+A producer raises synchronous events by calling the `fire()` method of the `Event`
 interface, passing the event object:
 
 [source.JAVA, java]
 -----------------------------
 documentEvent.fire(document);
 -----------------------------
 
-This particular event will be delivered to every observer method that:
+This particular event will only be delivered to synchronous observer method that:
 
 * has an event parameter to which the event object (the `Document`) is
 assignable, and
 * specifies no qualifiers.
 
-The container simply calls all the observer methods, passing the event
+The container simply calls all the synchronous observer methods, passing the event
 object as the value of the event parameter. If any observer method
 throws an exception, the container stops calling observer methods, and
 the exception is rethrown by the `fire()` method.
 
+==== Asynchronous event producers
+A producer raises asynchronous events by calling the `fireAsync()` method of the `Event`
+interface, passing the event object:
+
+[source.JAVA, java]
+-----------------------------
+documentEvent.fireAsync(document);
+-----------------------------
+
+This particular event will only be delivered to asynchronous observer method that:
+
+* has an event parameter to which the event object (the `Document`) is
+assignable, and
+* specifies no qualifiers.
+
+The container simply calls all the asynchronous observer methods, passing the event
```

Comment:
I think this should be more clear here. Ie. the method returns immediately and the observers are notified _"in one or more different threads"_... or something like that?